DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -2087,6 +2087,39 @@ private void BroadcastGame()
             sb.Append(";");
             sb.Append(Map?.SHA1);
 
+            List<IGameSessionSetting> broadcastableSettings = GetBroadcastableSettings();
+
+            List<int> gameOptionValues = new();
+
+            int checkboxCount = CheckBoxes.Count(cb => cb.BroadcastToLobby);
+            if (checkboxCount > 0)
+            {
+                bool[] checkboxValues = new bool[checkboxCount];
+                for (int i = 0; i < checkboxCount; i++)
+                    checkboxValues[i] = CheckBoxes.Where(cb => cb.BroadcastToLobby).ElementAt(i).Checked;
+
+                List<byte> byteList = Conversions.BoolArrayIntoBytes(checkboxValues).ToList();
+
+                // Pad to multiple of 4 bytes
+                while (byteList.Count % 4 != 0)
+                    byteList.Add(0);
+
+                byte[] byteArray = byteList.ToArray();
+
+                // Convert bytes to integers
+                for (int i = 0; i < byteArray.Length / 4; i++)
+                    gameOptionValues.Add(BitConverter.ToInt32(byteArray, i * 4));
```

Comment:
Same as the V3 PR: be careful using BitConverter. You can throw if `BitConverter.IsLittleEndian` does not match our expect, or use `BinaryPrimitives` instead

Reply:
Update: looks like we already have ~20 `BitConverter` usages. Maybe we should change them to `BinaryPrimitives` or globally throw on unexpected `BitConverter.IsLittleEndian` in another PR